xrpl/transaction/check_create.go

```diff
@@ -1,23 +1,82 @@
 package transaction
 
 import (
+	addresscodec "github.com/Peersyst/xrpl-go/address-codec"
 	"github.com/Peersyst/xrpl-go/xrpl/transaction/types"
 )
 
+// Create a Check object in the ledger, which is a deferred payment that can be cashed by its intended destination. The sender of this transaction is the sender of the Check.
+//
+// Example:
+//
+// ```json
+//
+//	{
+//	    "Account": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
+//	    "TransactionType": "EscrowCreate",
+//	    "Amount": "10000",
+//	    "Destination": "rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW",
+//	    "CancelAfter": 533257958,
+//	    "FinishAfter": 533171558,
+//	    "Condition": "A0258020E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855810100",
+//	    "DestinationTag": 23480,
+//	    "SourceTag": 11747
+//	}
+//
+// / ```
 type CheckCreate struct {
 	BaseTx
-	Destination    types.Address
-	SendMax        types.CurrencyAmount
-	DestinationTag uint32        `json:",omitempty"`
-	Expiration     uint32        `json:",omitempty"`
-	InvoiceID      types.Hash256 `json:",omitempty"`
+	// The unique address of the account that can cash the Check.
+	Destination types.Address
+	// Maximum amount of source currency the Check is allowed to debit the sender, including transfer fees on non-XRP currencies.
+	// The Check can only credit the destination with the same currency (from the same issuer, for non-XRP currencies). For non-XRP amounts, the nested field names MUST be lower-case.
+	SendMax types.CurrencyAmount
+	// (Optional) Arbitrary tag that identifies the reason for the Check, or a hosted recipient to pay.
+	DestinationTag uint32 `json:",omitempty"`
+	// (Optional) Time after which the Check is no longer valid, in seconds since the Ripple Epoch.
+	Expiration uint32 `json:",omitempty"`
```

Comment:
Add 0 by default

Reply:
I am not sure it's a good idea to set it to 0 by default as it's an optional field. It's possible to define the CheckCreate without it and setting it to 0 by default might confuse the user using the SDK.
What do you think @GuillemGarciaDev ?

Reply:
It is set by default if an integer variable is not initialized with a given value. It is a go language feature.